core/net/ip/resolv.c

```diff
@@ -954,7 +954,10 @@ newdata(void)
 #if !ARCH_DOESNT_NEED_ALIGNED_STRUCTS
     {
       static struct dns_answer aligned;
-      memcpy(&aligned, ans, sizeof(aligned));
+      uint8_t t;
```

Comment:
This workaround can be avoided if the structure dns_answer is defined with attribute "packed" as per below on line no. 232:

struct __attribute__ ((packed)) dns_answer {
  /* DNS answer record starts with either a domain name or a pointer
   * to a name already present somewhere in the packet. */
  uint16_t type;
  uint16_t class;
  uint16_t ttl[2];
  uint16_t len;
#if NETSTACK_CONF_WITH_IPV6
  uint8_t ipaddr[16];
#else
  uint8_t ipaddr[4];
#endif
};